source/geometry_model/spherical_chunk.cc

```diff
@@ -0,0 +1,416 @@
+      {
+        prm.enter_subsection("Chunk");
+        {
+          prm.declare_entry ("Chunk bottom radius", "0",
+                             Patterns::Double (0),
+                             "Radius at the bottom surface of the chunk. Units: m.");
+          prm.declare_entry ("Chunk top radius", "1",
+                             Patterns::Double (0),
+                             "Radius at the top surface of the chunk. Units: m.");
+
+          prm.declare_entry ("Chunk minimum longitude", "0",
+                             Patterns::Double (-180, 180),
+                             "Minimum longitude of the chunk. Units: m.");
+          prm.declare_entry ("Chunk maximum longitude", "1",
+                             Patterns::Double (-180, 180),
+                             "Maximum longitude of the chunk. Units: m.");
+
+          prm.declare_entry ("Chunk minimum latitude", "0",
+                             Patterns::Double (-90, 90),
+                             "Minimum latitude of the chunk. This value is ignored "
+                             "if the simulation is in 2d. Units: m.");
+          prm.declare_entry ("Chunk maximum latitude", "1",
+                             Patterns::Double (-90, 90),
+                             "Maximum latitude of the chunk. This value is ignored "
+                             "if the simulation is in 2d. Units: m.");
+
+          prm.declare_entry ("Radius repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in radius.");
+          prm.declare_entry ("Longitude repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in longitude.");
+          prm.declare_entry ("Latitude repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in latitude. This value is ignored "
+                             "if the simulation is in 2d");
+
+          prm.declare_entry ("Longitude periodic", "false",
+                             Patterns::Bool (),
+                             "Whether the box should be periodic in longitude.");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    SphericalChunk<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Spherical chunk");
+        {
+
+          const double degtorad = dealii::numbers::PI/180;
+
+          Assert (dim >= 1, ExcInternalError());
+          Assert (dim <= 3, ExcInternalError());
+
+          if (dim == 1)
+            {
+              point1[0] = prm.get_double ("Chunk bottom radius");
+              point2[0] = prm.get_double ("Chunk top radius");
+              repetitions[0] = prm.get_integer ("Radius repetitions");
+            }
+
+          if (dim == 2)
```

Comment:
But the order of inputs is different ... in 2D 
point*[1] =prm.get_double ("Chunk * radius");
in 3D:
point*[1] =prm.get_double ("Chunk * latitude");